fizzbuzz/src/main/java/fizz.buzz/number/factory/NumberFactoryImpl.java

```diff
@@ -1,23 +1,28 @@
 package fizz.buzz.number.factory;
 
-import fizz.buzz.number.*;
 import fizz.buzz.number.Number;
+import fizz.buzz.number.NumberFactory;
 
 public class NumberFactoryImpl implements NumberFactory {
-    public Number build(Integer input) {
 
-        if (input % 3 == 0 && input % 5 == 0) {
-            return new FizzBuzzNumber(input);
-        }
+    private FizzBuzzNumberFactory numberFactory;
 
-        if (input % 3 == 0) {
-            return new FizzNumber(input);
-        }
+    public NumberFactoryImpl() {
+        chainFactories();
+    }
 
-        if (input % 5 == 0) {
-            return new BuzzNumber(input);
-        }
+    private void chainFactories() {
+        FizzNumberFactory fizzFactory = new FizzNumberFactory();
+        BuzzNumberFactory buzzFactory = new BuzzNumberFactory();
+        RegularNumberFactory regularFactory = new RegularNumberFactory();
 
-        return new RegularNumber(input);
+        numberFactory = new FizzBuzzNumberFactory();
+        numberFactory.setSuccessor(fizzFactory);
+        fizzFactory.setSuccessor(buzzFactory);
+        buzzFactory.setSuccessor(regularFactory);
```

Comment:
Does it look any better, @tobias-reuter ? I tried to make the chaining more clear here.